{
  "cells": [
    {
      "cell_type": "raw",
      "metadata": {},
      "source": [
        "---\n",
        "title: \"Differential Expression and Annotation\"\n",
        "author: \"Benilton S Carvalho\"\n",
        "format: revealjs\n",
        "editor: visual\n",
        "---"
      ],
      "id": "fe872639"
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Tools for scRNA-Seq\n",
        "\n",
        "-   scanpy\n",
        "-   Monocle\n",
        "-   Bioconductor\n",
        "-   Seurat\n",
        "\n",
        "# Differential Expression\n",
        "\n",
        "## Methods\n",
        "\n",
        "-   Preprocessing: data import, QC, quantification\n",
        "-   Normalization: dropout, depth, batch effects\n",
        "-   Dimensionality Reduction: PCA, UMAP\n",
        "-   Clustering: cell type clusters\n",
        "-   Differential Gene Expression\n",
        "\n",
        "## Seurat\n",
        "\n",
        "::: columns\n",
        "::: {.column width=\"50%\"}\n",
        "-   Most used tool\n",
        "\n",
        "-   Good documentation\n",
        "\n",
        "-   Several tutorials\n",
        "\n",
        "-   Many methods\n",
        "\n",
        "-   Extendable\n",
        "\n",
        "-   Gene Expression Dynamics\n",
        ":::\n",
        "\n",
        "::: {.column width=\"50%\"}\n",
        "![](umap.png)\n",
        ":::\n",
        ":::\n",
        "\n",
        "## Statistical Analysis of Differences Between Clusters\n",
        "\n",
        "-   Different types of hits\n",
        "    -   Quantitatively significant between clusters\n",
        "    -   Qualitatively different (predictive) of cluster membership\n",
        "-   Different types of markers\n",
        "    -   Global: Distinguish one cluster from all of the rest of the data\n",
        "    -   Local: Distinguish one cluster from another defined set of clusters\n",
        "\n",
        "## Statistical Analysis of Differences Between Clusters\n",
        "\n",
        "-   Often filter genes based on coverage in the set or the size of groups\n",
        "-   Several choices of methods to identify genes\n",
        "\n",
        "## Methods\n",
        "\n",
        "-   Non-parametric: Wilcoxon rank sum test\n",
        "-   Parametric: t-test, negative binomial\n",
        "-   Classification: ROC\n",
        "-   Specialized: MAST\n",
        "\n",
        "```         \n",
        "FindMarkers(data, ident.1 = \"g1\", ident.2 = \"g2\", group.by = \"status\", test.use = \"roc\", only.pos = TRUE)\n",
        "```\n",
        "\n",
        "## \n",
        "\n",
        "![](methods.png){fig-align=\"center\"}\n",
        "\n",
        "# Cell Type Annotation\n",
        "\n",
        "## Why annotate cell types?\n",
        "\n",
        "* Interpreting the findings of our analysis is the most difficult task in sc-data analysis\n",
        "* Understanding the biological state of each cluster is way harder then assigning clusters\n",
        "* To do this, we need to “connect” our dataset to existing knowledge\n",
        "* One strategy is to compare the expression of our dataset to the expressions of curated existing datasets (references)\n",
        "* What tool do we use? **SingleR**\n",
        "\n",
        "## Cell Type Annotation\n",
        "\n",
        "* **SingleR** pkg contains the statistical method for assignment\n",
        "* **celldex** pkg shares several reference (well curated) datasets\n",
        "* Most references are built from bulk RNA-seq and microarray\n",
        "* They are good enough for annotation of sc-data, provided that the references contains the cell types that are expected to be present on the test data\n",
        "* We'll use a reference built from Blueprint and ENCODE data\n",
        "* Single-cell references can also be used (eg. `MuraroPancreasData()`)\n",
        "\n",
        "## How to perform annotation?\n",
        "\n",
        "```\n",
        "## Load the references\n",
        "library(celldex)\n",
        "ref = BlueprintEncodeData()\n",
        "\n",
        "## Compare expression levels from my.data to the reference\n",
        "library(SingleR)\n",
        "pred = SingleR(test = my.data, ref = ref, labels = ref$label.main)\n",
        "\n",
        "table(pred$labels)\n",
        "```\n",
        "\n",
        "\n",
        "```{css}\n",
        "code.sourceCode {\n",
        "  font-size: 1.3em;\n",
        "  /* or try font-size: xx-large; */\n",
        "}\n",
        "```"
      ],
      "id": "df71a847"
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}